ocp_on_osp.yml

```diff
@@ -0,0 +1,125 @@
+
+    - name: set quotas to unlimited
+      shell: |
+        source /home/stack/overcloudrc
+        openstack quota set --properties -1 --server-groups -1 --ram -1 --key-pairs -1 --instances -1  --cores -1 --per-volume-gigabytes -1 --gigabytes -1 --backup-gigabytes -1 --snapshots -1 --volumes -1 --backups -1 --subnetpools -1 --ports -1 --subnets -1 --networks -1 --floating-ips -1 --secgroup-rules -1 --secgroups -1 --routers -1 --rbac-policies -1 {{ shiftstack_id.stdout }}
+      args:
+        chdir: /home/stack
+
+    - name: create shiftstackrc
+      shell: |
+        sed -e 's/OS_USERNAME=.*/OS_USERNAME=shiftstack_user/' \
+        -e 's/OS_PROJECT_NAME=.*/OS_PROJECT_NAME=shiftstack/' \
+        -e 's/OS_CLOUDNAME=.*/OS_CLOUDNAME=shiftstack/' \
+        -e 's/OS_PASSWORD=.*/OS_PASSWORD=redhat/' overcloudrc > shiftstackrc
+
+    - name: get cloud.yaml variables
+      shell: |
+        for key in $( set | awk '{FS="="}  /^OS_/ {print $1}' ); do unset $key ; done
+        source /home/stack/shiftstackrc
+        echo -n " shiftstack: \
+          {'auth': \
+            { 'auth_url': '$OS_AUTH_URL', \
+              'username': '$OS_USERNAME', \
+              'password': '$OS_PASSWORD', \
+              $(if [ -n "$OS_USER_DOMAIN_NAME" ]; then echo "'user_domain_name': '${OS_USER_DOMAIN_NAME}',"; fi) \
+              $(if [ -n "$OS_PROJECT_DOMAIN_NAME" ]; then echo "'project_domain_name': '${OS_PROJECT_DOMAIN_NAME}',"; fi) \
+              'project_name': '${OS_PROJECT_NAME:-$OS_TENANT_NAME}' \
+            } $(if [ -n "$OS_IDENTITY_API_VERSION" ]; then echo ",'identity_api_version': '${OS_IDENTITY_API_VERSION}'"; fi) }"
+      register: cloud_details
+
+    - name: Set clouds_yaml fact
+      set_fact:
+        clouds_yaml: "{{ cloud_details.stdout|from_yaml }}" 
+
+    - name: insert cloud parameters
+      blockinfile:
+        dest: "~/clouds.yaml"
+        block: |5
+              {{ clouds_yaml|to_nice_yaml(indent=4) }}
+        insertbefore: "#END shiftstack SECTION"
+        marker: "#{mark} shiftstack PARAMETERS"
+
+    - name: create floating ip
+      shell: |
+        source /home/stack/overcloudrc
+        neutron floatingip-create --floating-ip-address 172.17.5.218 --tenant-id {{ shiftstack_id.stdout }} nova
+      ignore_errors: true
+
+    - name: add DNS detail in etc/hosts
+      blockinfile:
+        path: "/etc/hosts"
+        block: |
+            172.17.5.218 api.ostest.shiftstack.com
+            172.17.5.218 console-openshift-console.apps.ostest.shiftstack.com
+            172.17.5.218 openshift-authentication-openshift-authentication.apps.ostest.shiftstack.com
+        insertafter: "EOF"
+      become: yes
+      become_user: root
+
+    - name: create install dir
+      file:
+        path: /home/stack/ostest
+        state: directory
+
+    - name: prepare install-config
+      template:
+        src: "{{ playbook_dir }}/templates/install-config.yaml.j2"
+        dest: "/home/stack/ostest/install-config.yaml"
```

Comment:
create install-config.yaml in /home/stack and add a next ansible task which copies it to /home/stack/ostest.

When we run "destroy cluster", it will remove /home/stack/ostest/install-config.yaml. So having a copy in /home/stack is always better.